frontend/src/Editor/Components/Datepicker.jsx

```diff
@@ -36,10 +36,11 @@ export const Datepicker = function Datepicker({
   };
 
   const onDateChange = (date) => {
-    fireEvent('onSelect');
     setDate(date);
     const dateString = computeDateString(date);
-    setExposedVariable('value', dateString);
+    setExposedVariable('value', dateString).then(() => {
+      fireEvent('onSelect');
```

Comment:
@manishkushare , letting you know that we fire the event on the then function, not on the next line.